src/par_iter/mod.rs

```diff
@@ -363,6 +365,16 @@ pub trait ParallelIterator: Sized {
         reduce(self, reduce_op)
     }
 
+    /// Takes two iterators and creates a new iterator over both.
+    ///
+    /// As with most parallel iterators, the order in which these are
+    /// combined is completely unspecified.
```

Comment:
Ah right.  For things like `enumerate`, `zip`, and `collect`, the items are ordered `a` then `b`.  I was just trying to parrot the normal parallel hazard that there's no *temporal* order to visitation.  Maybe I should s/combined/visited/ ?